### PR TITLE
5/7 test cases for I/O asserts

### PR DESCRIPTION
**Pull request by AdamMutzell**
Started using the Hypothesis library for fuzzing. This is related to issue #15 , where this one solves the first five. There are several failures generated that I have not looked into.

### CODE REVIEW

**Comment by Uplink036** [comment]
Don't feel like merging until the errors are dealt with. Meaning finding why they occur and trying to fix that, if we can't find a fix then we might need to discuss again.

File: test_pickle.py [hunk 1]
```diff
@@ -7,4 +24,101 @@ def test_pickle(self):
...
+
+        assert h_a1 == h_a2
+        assert h_b1 == h_b2
+        assert h_c1 == h_c2
+        
+
+    #Maybe consider utf-16
+    #given strings, bytes, bytearrays (from ints is what I've done)
+    @given(st.text(), st.binary(), st.lists(st.integers(min_value=0, max_value=255)))
+    def test_strings(self, a, b, c):
```
Comment (Uplink036) [inline test_pickle.py:63]: Why can't this function be be broken up into 4 different ones?

File: test_pickle.py [hunk 2]
```diff
@@ -7,4 +24,101 @@ def test_pickle(self):
...
+        assert h_a1 == h_a2
+        assert h_b1 == h_b2
+        assert h_c1 == h_c2
+
+
+    @given(st.tuples(st.integers(), st.floats(), st.text(), st.binary(), st.booleans()),
+           st.lists(elements=one_dim_random_strategy()),
+           st.sets(elements=one_dim_random_strategy()),
+           st.dictionaries(keys=st.text(), values=one_dim_random_strategy()))
+    def test_collections(self, a, b, c, d):
```
Comment (Uplink036) [inline test_pickle.py:90]: Again, why can't this be broken up?

**Comment by Uplink036** [comment]
See above comments, we don't need to change but I have to ask. Otherwise it's good enough to merge according to me.

File: test_pickle.py (hunk 2 above)
Comment (AdamMutzell) [reply test_pickle.py:90]: We can discuss!

**Comment by Uplink036** [comment]
Looks good to me, all discussion above seems to have been fixed.  We can merge it now.

**Pull request merged**